microcosm_pubsub/consumer.py

```diff
@@ -142,6 +95,56 @@ def nack(self, message):
         """
         pass
 
+    def _from_sqs(self, sqs_message, validate_md5=False):
+        """
+        Create an `SQSMessage` from SQS data.
+
+        """
+        message_id = sqs_message["MessageId"]
+        receipt_handle = sqs_message["ReceiptHandle"]
+        body = sqs_message["Body"]
+
+        if validate_md5:
+            self._validate_md5(sqs_message, body)
+
+        message = loads(body)["Message"]
+        media_type, content = self._parse_message_from_sqs(message)
```

Comment:
We parse the media type and content here. If we want to support other kind of message enveloping (e.g. using SQS metadata) or bypass enveloping (using a raw message) with a vanilla media type of `application/json`, we just need to swap out this function.

The next step would be to build an "enveloping" module and have this function chose based on config.